### PR TITLE
Add clipboard button to the Codeblock

### PR DESCRIPTION
**Pull request by FMMazur**
Solve: #2657

### CODE REVIEW

File: packages/editor/src/extensions/code-block/component.tsx [hunk 1]
```diff
@@ -27,22 +27,61 @@ import { CodeBlockAttributes } from "./code-block";
...
+    if (isNotSelecting) {
+      editor.commands.setTextSelection({
+        from: codeblock.pos + 1,
+        to: finalPosition
+      });
+    }
+    editor.commands.focus();
+    editor.commands.copy(codeblock.node.textContent);
+    start();
+  };
```
Comment (thecodrr) [inline packages/editor/src/extensions/code-block/component.tsx:78]: This all is unnecessary. Inline the onClick handler code & just call `writeText(node.textContent)`.

File: packages/editor/src/extensions/code-block/code-block.ts [hunk 2]
```diff
@@ -276,7 +287,10 @@ export const CodeBlock = Node.create<CodeBlockOptions>({
             indentLength: options.amount
           });
           return true;
-        }
+        },
+      copy: (text) => () => {
+        return this.options.onClickToCopy(text);
+      }
```
Comment (thecodrr) [inline packages/editor/src/extensions/code-block/code-block.ts:293]: There is no need for a new `copy` command on the editor for abstracting the copy command to the clients because the `writeText` from `clipboard-polyfill` works on all platforms automatically.

File: packages/editor/src/extensions/code-block/component.tsx [hunk 3]
```diff
@@ -27,22 +27,61 @@ import { CodeBlockAttributes } from "./code-block";
...
+
+  const copyButtonStyle = {
+    ...(enabled && { borderColor: "primary" }),
+    ...(!isMobile && { "div:hover > &": { opacity: 1 } })
+  };
```
Comment (thecodrr) [inline packages/editor/src/extensions/code-block/component.tsx:83]: Why is this here instead of inlined where its needed?

File: packages/editor/src/extensions/code-block/component.tsx [hunk 4]
```diff
@@ -144,6 +183,34 @@ export function CodeblockComponent(
             </Text>
           </Button>
         </Flex>
+
+        {node.textContent.length > 0 ? (
+          <Button
+            variant={"tool"}
+            sx={{
+              position: "absolute",
+              opacity: Number(isMobile),
```
Comment (thecodrr) [inline packages/editor/src/extensions/code-block/component.tsx:192]: Let's be obvious and say `isMobile ? 1 : 0`.

File: packages/editor/src/extensions/code-block/highlighter.ts [hunk 5]
```diff
@@ -181,7 +183,11 @@ export function HighlighterPlugin({
         const languages: Record<string, string> = {};
         findChildren(state.doc, (node) => node.type.name === name).forEach(
           (block) => {
-            const { id, language } = block.node.attrs;
+            const attrs = block.node.attrs as Writeable<Attrs>;
+            const lines = toCodeLines(block.node.textContent, block.pos);
+            attrs.lines = lines.slice();
+
+            const { id, language } = attrs;
```
Comment (thecodrr) [inline packages/editor/src/extensions/code-block/highlighter.ts:190]: This change is not needed.

**Review by thecodrr — CHANGES_REQUESTED** [review]
Hey! Thanks for this. However, there are a few changes necessary.

File: packages/editor/src/extensions/code-block/code-block.ts (hunk 2 above)
Comment (thecodrr) [reply packages/editor/src/extensions/code-block/code-block.ts:293]: Yes, that'd be best. The editor gets rendered inside a webview so clipboard-polyfill works there as well.

File: packages/editor/src/extensions/code-block/component.tsx (hunk 3 above)
Comment (thecodrr) [reply packages/editor/src/extensions/code-block/component.tsx:83]: In your case, yes but rewrite it using simple ternaries:

```jsx
borderColor: enabled ? "primary" : "transparent",
"div:hover > &": isMobile ? {} : { opacity: 1 }
```

**Comment by thecodrr** [comment]
@FMMazur what's the status on this?

**Comment by FMMazur** [comment]
> @FMMazur what's the status on this?

I didn't have time to touch it during the week, as it was just removing stuff I did and tested today, at least the web and desktop version, I couldn't test on mobile.

**Comment by thecodrr** [comment]
CI is still complaining about `Missing: clipboard-polyfill@4.0.0 from lock file`.
